### PR TITLE
Fingerprint sensors

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

**Comment by Mic92** [comment]
@r-vdp does this work for you?

```
nix-repl> :p nixosConfigurations.turingmachine.config.facter.detected.fingerprint
{ enable = true; }
```

File: modules/nixos/fingerprint/devices.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
+{
+    "045E:00BB": true,
```
Comment (Mic92) [inline modules/nixos/fingerprint/devices.json:2]: I have the feeling we should just store vendor and product ids as hex strings actually as this the more common format.

Comment (brianmcgee) [reply modules/nixos/fingerprint/devices.json:2]: Do you mean output them as hex in the report?

Comment (Mic92) [reply modules/nixos/fingerprint/devices.json:2]: Yes.

Comment (Mic92) [reply modules/nixos/fingerprint/devices.json:2]: It's more readable since everyone is using this and I have to convert to hex all the time.

Comment (Mic92) [reply modules/nixos/fingerprint/devices.json:2]: We should also do lower case hex because this is what is mostly used. `lib.toHexString` does uppercase.

Comment (Mic92) [reply modules/nixos/fingerprint/devices.json:2]: Ah and I just noticed it has to be zero-padded as well.

**Comment by Mic92** [comment]
I tested this now on two different laptops. I thing this is good to go for the first iteration.

**Comment by Mic92** [comment]
@mergify queue

**Comment by Mic92** [comment]
@mergify refresh

**Comment by Mic92** [comment]
@r-vdp we have too many frameworks. Not enough diversity.

**Comment by brianmcgee** [comment]
> @r-vdp we have too many frameworks. Not enough diversity.

Are you saying I need to expense some hardware? :thinking:

**Comment by Mic92** [comment]
@brianmcgee you probably have to travel to thailand some time.

**Pull request merged**